### PR TITLE
feat(rln-relay): adds rln identifier to the RateLimitProof

### PR DESCRIPTION
**Pull request by staheri14**
Addresses https://github.com/status-im/nwaku/issues/1213. 
It also enables compilation of waku-rln-relay when building wakunode2. Without this, waku v2 test fleets will not be able to run rln-relay protocol with zero-kit.

### CODE REVIEW

File: waku/v2/node/wakunode2.nim [hunk 1]
```diff
@@ -36,7 +36,7 @@ import
 export
   wakunode2_types
 
-when defined(rln):
+when defined(rln) or defined(rlnzerokit):
```
[the remark this answers is not in the document]
Comment (staheri14) [reply waku/v2/node/wakunode2.nim:39]: That was only tested in chat2 but not waku v2/

File: waku/v2/protocol/waku_rln_relay/waku_rln_relay_types.nim [hunk 2]
```diff
@@ -32,14 +32,15 @@ type
...
   Epoch* = array[32, byte]
+  RlnIdentifier* = array[32, byte]
 
 when defined(rln) or (not defined(rln) and not defined(rlnzerokit)):
   type
     ZKSNARK* = array[256, byte]
 when defined(rlnzerokit):
   type 
     ZKSNARK* = array[128, byte]
-    RlnIdentifier* = array[32, byte]
```
[the remark this answers is not in the document]
Comment (staheri14) [reply waku/v2/protocol/waku_rln_relay/waku_rln_relay_types.nim:42]: You are right, they only differ in the `proof` field, but that is already moved under a compiler flag. Going to unify the two type defs.

[the remark this answers is not in the document]
Comment (staheri14) [reply waku/v2/protocol/waku_rln_relay/waku_rln_relay_types.nim:42]: Done https://github.com/status-im/nwaku/pull/1219/commits/867fbb064759d398bd35d70f9196a0c858232449

**Comment by staheri14** [comment]
> How we were able to successfully test zerokit before if some flags in wakunode2 were missing?

Our previous tests were for chat2. Other tests were also targeted on specific procs, which is why everything was ok. But, we had not tested mounting rln-relay in wakunode2.

**Review by rymnc — APPROVED** [review]
LGTM, thanks! 😁

**Review by jm-clius — APPROVED** [review]

**Review by richard-ramos — APPROVED** [review]

**Pull request merged**